src/e2e/java/teammates/e2e/util/JMeterConfig.java

```diff
@@ -122,26 +135,64 @@ public HashTree createTestPlan() {
         onceOnlyController.setProperty(TestElement.TEST_CLASS, OnceOnlyController.class.getName());
         onceOnlyController.setProperty(TestElement.GUI_CLASS, OnceOnlyControllerGui.class.getName());
 
+        // Regex Extractor for CSRF token
+        RegexExtractor regexExtractor = null;
+        if (isPostEndpoint) {
+            regexExtractor = new RegexExtractor();
+            regexExtractor.setName("Regular Expression Extractor");
+            regexExtractor.setUseField("true"); // Find matches in response headers
+            regexExtractor.setRefName("csrfToken");
+            regexExtractor.setRegex("Set-Cookie: CSRF-TOKEN=(.+?);");
+            regexExtractor.setTemplate("$1$");
+            regexExtractor.setProperty(TestElement.TEST_CLASS, RegexExtractor.class.getName());
+            regexExtractor.setProperty(TestElement.GUI_CLASS, RegexExtractorGui.class.getName());
+        }
+
         // Test API Endpoint HTTP Request
         HTTPSamplerProxy apiSampler = new HTTPSamplerProxy();
         apiSampler.setName("Test Endpoint");
         apiSampler.setPath(getTestEndpoint());
         apiSampler.setMethod(getTestMethod());
-        getTestArguments().forEach((key, value) -> {
+        getTestEndpointParameters().forEach((key, value) -> {
             apiSampler.addArgument(key, value);
         });
         apiSampler.setEnabled(true);
         apiSampler.setProperty(TestElement.TEST_CLASS, HTTPSamplerProxy.class.getName());
         apiSampler.setProperty(TestElement.GUI_CLASS, HttpTestSampleGui.class.getName());
 
+        // Add HTTP POST Body Data
+        if (isPostEndpoint && !getTestEndpointPostVariables().isEmpty()) {
```

Comment:
This comment is likely not useful, but there is no need for the empty check because if the list is empty, the subsequent for loop is equivalent to doing nothing anyway.